pkgs/applications/networking/localsend/default.nix

```diff
@@ -1,49 +1,80 @@
-{ lib, stdenv, appimageTools, fetchurl, undmg }:
+{ lib
+, stdenv
+, fetchurl
+, fetchFromGitHub
+, flutter
+, makeDesktopItem
+, pkg-config
+, libayatana-appindicator
+, undmg
+}:
 
 let
   pname = "localsend";
-  version = "1.11.1";
+  version = "1.12.0";
 
-  hashes = {
-    x86_64-linux = "sha256-K4M9cks0FNsCLIqQhSgUAz3tRMKng6JkZ/ZfwG2hZJA=";
-    x86_64-darwin = "sha256-Cixo00I4BBAmUnszsz+CxPX3EY175UTufCmwQmIsEgg=";
-  };
+  linux = flutter.buildFlutterApplication {
+    inherit pname version;
 
-  srcs = rec {
-    x86_64-linux = fetchurl {
-      url = "https://github.com/localsend/localsend/releases/download/v${version}/LocalSend-${version}-linux-x86-64.AppImage";
-      hash = hashes.x86_64-linux;
-    };
-    x86_64-darwin = fetchurl {
-      url = "https://github.com/localsend/localsend/releases/download/v${version}/LocalSend-${version}.dmg";
-      hash = hashes.x86_64-darwin;
+    src = fetchFromGitHub {
+      owner = pname;
+      repo = pname;
+      rev = "v${version}";
+      hash = "sha256-mk0CLZP0x/mEixeAig7X41aFgQzs+kZkBJx6T//3ZKY=";
     };
-    aarch64-darwin = x86_64-darwin;
-  };
-  src = srcs.${stdenv.hostPlatform.system} or (throw "Unsupported system for package localsend: ${stdenv.hostPlatform.system}");
 
-  appimageContents = appimageTools.extract { inherit pname version src; };
+    sourceRoot = "source/app";
+    depsListFile = ./deps.json;
+    vendorHash = "sha256-fXzxT7KBi/WT2A5PEIx+B+UG4HWEbMPMsashVQsXdmU=";
 
-  linux = appimageTools.wrapType2 rec {
-    inherit pname version src meta;
+    nativeBuildInputs = [ pkg-config ];
 
-    extraPkgs = p: [ p.ayatana-ido p.libayatana-appindicator p.libayatana-indicator p.libdbusmenu p.libepoxy ];
+    buildInputs = [ libayatana-appindicator.dev ];
 
-    extraInstallCommands = ''
-      mv $out/bin/${pname}-${version} $out/bin/${pname}
-
-      install -m 444 -D ${appimageContents}/org.localsend.localsend_app.desktop \
-        $out/share/applications/${pname}.desktop
-      substituteInPlace $out/share/applications/${pname}.desktop \
-        --replace 'Exec=localsend_app' "Exec=$out/bin/localsend"
+    preBuild = ''
+      doPubGet flutter pub get --offline -v
+      # $PUB_CACHE/hosted is a symlink to a store path.
+      mv $PUB_CACHE/hosted $PUB_CACHE/hosted_copy
+      cp -HR $PUB_CACHE/hosted_copy $PUB_CACHE/hosted
+      substituteInPlace $PUB_CACHE/hosted/pub.dev/system_tray-*/linux/tray.cc \
```

Comment:
This should be put in a [package override](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/dart/package-overrides).

#263345 will change how this works, though. I suggest waiting until that goes through, and then a nicer solution can be made.

Reply:
> This should be put in a [package override](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/compilers/dart/package-overrides).

Thanks! Is there any documents?

> https://github.com/NixOS/nixpkgs/pull/263345 will change how this works, though. I suggest waiting until that goes through, and then a nicer solution can be made.

I'd like to wait for it if it can be merged in a week. I don't want to wait several months.

Reply:
I moved the hack to package override. It's still hacky.